cpp/src/arrow/util/bit_stream_utils.h

```diff
@@ -398,6 +412,27 @@ inline int BitReader::GetBatch(int num_bits, T* v, int batch_size) {
   return batch_size;
 }
 
+#ifdef ARROW_WITH_QPL
+inline bool BitReader::GetBatchWithQpl(int batch_size, qpl_job* job) {
+  if (!job) {
+    return false;
+  }
+  job->param_low = value_offset_;
+  job->param_high = batch_size + value_offset_;
+  job->num_input_elements = batch_size + value_offset_;
+
+  job->next_in_ptr = const_cast<uint8_t*>(buffer_ - 1);
```

Comment:
It will not touch invlalid memory since in function  [DictDecoderImpl::SetData](https://github.com/apache/arrow/blob/f6685371032e933e31d7ae2b156a6385c00d1673/cpp/src/parquet/encoding.cc#L1491), bit reader buffer was set to <encoded-data> and skip <bit_width>
![image](https://user-images.githubusercontent.com/100556669/207534289-0ebf8d42-dccc-4d3e-9131-24efc9c32ed6.png)

IAA need a full parquet rle buffer so I `-1` here

Reply:
Oh, does it mean that we can use this feature only for Parquet's RLE format?
Could you tell me IAA's document for this?

Reply:
Or is "bit-width" included in general RLE format?